core/src/main/scala/cats/data/OneAnd.scala

```diff
@@ -123,8 +123,21 @@ final case class OneAnd[F[_], A](head: A, tail: F[A]) {
     s"OneAnd(${A.show(head)}, ${FA.show(tail)})"
 }
 
+private[data] trait OneAndBinCompat0[F[_], A] {
+  val head: A
+  val tail: F[A]
```

Comment:
Abstract vals look suspicious a bit. And might not be necessary – since `OneAndBinCompat0` is supposed to be extended by `OneAnd` only, a trick with self-type annotation might work here:
```scala
private[data] trait OneAndBinCompat0[F[_], A] {
  self: OneAnd[F, A] => // allow access to `head` and `tail` from withing this trait
```

Reply:
For some reason I've never used a non-`trait` type for self-type annotation and couldn't come up with this!
Applied: af23366